File: Python/isPowerOfTwo.py
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        return n > 0 and (n & (n - 1)) == 0

def main():
    solution = Solution()
    result = solution.isPowerOfTwo(1)
    result2 = solution.isPowerOfTwo(16)
    result3 = solution.isPowerOfTwo(0)
    print(result)
    print(result2)
    print(result3)

if __name__ == "__main__":
    main()

File: C++/isPowerOfTwo.cpp
#include <iostream>
#include <string>
using namespace std;

// class Solution {
//     public:
//         bool isPowerOfTwo(int n) {
//             if (n <= 0) return false;
//             while (n > 1) {
//                 if (n % 2 != 0) return false;
//                 n /= 2;
//             }
//             return true;
//         }
//     };

// Deepseek:
// n = 8，二进制是 1000。
// n - 1 = 7，二进制是 0111。
// 1000 & 0111 = 0000。
// 返回值：
// 如果 (n & (n - 1)) == 0，返回 true（或 True），否则返回 false（或 False）。
class Solution {
    public:
        bool isPowerOfTwo(int n) {
            if (n <= 0) return false;
            return (n & (n - 1)) == 0;
        }
    };

int main() {
    Solution solution;
    cout << solution.isPowerOfTwo(1) << endl;
    cout << solution.isPowerOfTwo(16) << endl;
    cout << solution.isPowerOfTwo(3) << endl;
    return 0;
}

